gcloud-java-compute/src/main/java/com/google/gcloud/compute/Operation.java

```diff
@@ -0,0 +1,793 @@
+    this.startTime = builder.startTime;
+    this.endTime = builder.endTime;
+    this.errors = builder.errors != null ? ImmutableList.copyOf(builder.errors) : null;
+    this.warnings = builder.warnings != null ? ImmutableList.copyOf(builder.warnings) : null;
+    this.httpErrorStatusCode = builder.httpErrorStatusCode;
+    this.httpErrorMessage = builder.httpErrorMessage;
+    this.description = builder.description;
+  }
+
+  /**
+   * Returns the operation's {@code Compute} object used to issue requests.
+   */
+  public Compute compute() {
+    return compute;
+  }
+
+  /**
+   * Returns the service-defined unique identifier for the operation.
+   */
+  public String id() {
+    return id;
+  }
+
+  /**
+   * Returns the creation timestamp in milliseconds since epoch.
+   */
+  public Long creationTimestamp() {
+    return creationTimestamp;
+  }
+
+  /**
+   * Returns the operation's identity. This method returns an {@link GlobalOperationId} for global
+   * operations, a {@link RegionOperationId} for region operations and a {@link ZoneOperationId} for
+   * zone operations.
+   *
+   * @see <a href="https://www.ietf.org/rfc/rfc1035.txt">RFC1035</a>
+   */
+  @SuppressWarnings("unchecked")
+  public <T extends OperationId> T operationId() {
+    return (T) operationId;
+  }
+
+  /**
+   * Reserved for future use.
+   */
+  String clientOperationId() {
+    return clientOperationId;
+  }
+
+  /**
+   * Returns the type of operation.
+   */
+  public String operationType() {
+    return operationType;
+  }
+
+  /**
+   * Returns the URL of the resource that the operation is modifying.
+   */
+  public String targetLink() {
+    return targetLink;
+  }
+
+  /**
+   * Returns the unique service-defined target ID, which identifies the resource that the operation
+   * is modifying.
+   */
+  public String targetId() {
+    return targetId;
+  }
+
+  /**
+   * Returns the status of the operation.
+   */
+  public Status status() {
+    return status;
+  }
+
+  /**
+   * Returns an optional textual description of the current status of the operation.
+   */
+  public String statusMessage() {
+    return statusMessage;
+  }
+
+  /**
+   * Returns the user who requested the operation, for example: {@code user@example.com}.
+   */
+  public String user() {
+    return user;
+  }
+
+  /**
+   * Returns an optional progress indicator that ranges from 0 to 100. There is no requirement that
+   * this be linear or support any granularity of operations. This should not be used to guess when
+   * the operation will be complete. This number should monotonically increase as the operation
+   * progresses.
+   */
+  public Integer progress() {
+    return progress;
+  }
+
+  /**
+   * Returns the time that this operation was requested. In milliseconds since epoch.
+   */
+  public Long insertTime() {
+    return insertTime;
+  }
+
+  /**
+   * Returns the time that this operation was started by the service. In milliseconds since epoch.
```

Comment:
should we make it explicit stating that it would be null if not started and bellow it would be null if not ended?